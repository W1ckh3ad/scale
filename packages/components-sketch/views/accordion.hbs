<div data-sketch-artboard="Accordion" class="artboard">
    <h1 class="title--artboard">Accordion</h1>

    <h2 class="title--artboard-section">Agent States</h2>
    <div class="agent-states--grid">
        <div class="agent-states--item">
          <p class="agent-states--label">:available</p>
          <scale-accordion>
            <scale-collapsible>
              <scale-text slot="headline" variant="collapsible">Type something 1</scale-text>
            </scale-collapsible>
          </scale-accordion>
        </div>
        <div class="agent-states--item">
          <p class="agent-states--label">:hover</p>
          <scale-accordion data-sketch-state="hover">
            <scale-collapsible is-expanded>
              <scale-text slot="headline" variant="collapsible">Type something 2</scale-text>
            </scale-collapsible>
          </scale-accordion>
        </div>
        <div class="agent-states--item">
          <p class="agent-states--label">:focus</p>
          <scale-accordion data-sketch-state="focus">
            <scale-collapsible is-expanded>
              <scale-text slot="headline" variant="collapsible">Type something 3</scale-text>
            </scale-collapsible>
          </scale-accordion>
        </div>
        <div class="agent-states--item">
          <p class="agent-states--label">:pressed</p>
          <scale-accordion data-sketch-state="active">
            <scale-collapsible is-expanded>
              <scale-text slot="headline" variant="collapsible">Type something 4</scale-text>
            </scale-collapsible>
          </scale-accordion>
        </div>
        <div class="agent-states--item">
          <p class="agent-states--label">:disabled</p>
          â€“
        </div>
    </div>
</div>
