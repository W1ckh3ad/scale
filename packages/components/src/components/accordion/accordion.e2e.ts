import { newE2EPage } from '@stencil/core/testing';

describe('scale-alert', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-alert>Notification</scale-alert>');
    const element = await page.find('scale-alert');
    expect(element).toHaveClass('hydrated');
  });
});
