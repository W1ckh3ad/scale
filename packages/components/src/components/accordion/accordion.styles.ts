import { JssStyle } from 'jss';

export const styles: JssStyle = {
  accordion: {
    margin: 0,

    padding: 0,
    width: '20em',
  },
  accordionContainer: { borderRadius: '7px', marginBottom: '1em' },
  accordionTrigger: {
    background: 'none',
    color: 'hsl(0, 0%, 13%)',
    display: 'block',
    fontSize: '1rem',
    fontWeight: 'normal',
    margin: 0,
    padding: '1em 1.5em',
    position: 'relative',
    textAlign: 'left',
    width: '100%',
    outline: 'none',
    borderStyle: 'none',
    '&:focus': {
      background: 'hsl(216, 94%, 94%)',
    },
    '&:hover': {
      background: 'hsl(216, 94%, 94%)',
    },
    '&:focus .accordionTitle': {
      borderColor: 'hsl(216, 94%, 73%)',
    },
    '&:focus .accordionIcon': {
      borderColor: 'hsl(216, 94%, 73%)',
    },
    '&[aria-expanded="true"] .accordionIcon': {
      transform: 'translateY(-50%) rotate(-135deg)',
    },
  },
  accordionTitle: {
    display: 'block',
    pointerEvents: 'none',
    border: 'transparent 2px solid',
    borderRadius: '5px',
    padding: '0.25em',
    outline: 'none',
  },
  accordionIcon: {
    border: 'solid hsl(0, 0%, 62%)',
    borderWidth: '0 2px 2px 0',
    height: '0.5rem',
    pointerEvents: 'none',
    position: 'absolute',
    right: '2em',
    top: '50%',
    width: '0.5rem',
    transform: 'translateY(-5px) rotate(45deg)',
  },
  accordionPanel: {
    margin: 0,
    padding: '1em 1.5em',
  },
};
