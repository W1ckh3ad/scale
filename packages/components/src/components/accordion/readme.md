# scale-accordion



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute   | Description | Type                                     | Default     |
| ----------- | ----------- | ----------- | ---------------------------------------- | ----------- |
| `dependend` | `dependend` |             | `boolean`                                | `false`     |
| `headline`  | `headline`  |             | `string`                                 | `undefined` |
| `styles`    | --          |             | `StyleSheet<string \| number \| symbol>` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
