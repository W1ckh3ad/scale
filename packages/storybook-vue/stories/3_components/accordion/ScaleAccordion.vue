<template>
  <scale-accordion :styles="styles" :dependent="dependent" :expanded="expanded">
    <slot></slot>
  </scale-accordion>
</template>

<script>
export default {
  props: {
    styles: { type: String },
    dependent: { type: Boolean, default: false },
    expanded: { type: Boolean, default: false },
  },
};
</script>