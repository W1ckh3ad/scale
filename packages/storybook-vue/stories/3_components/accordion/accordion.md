# Accordion

Accordions make it easy for users to get an overview of page content and expand the relevant topics for further exploration. Accordions can hold all types of content.

## General

### When to Use

- To organize related information.
- To shorten pages and reduce scrolling when it is not crucial to read content in full.
- When space is at a premium and long content cannot be displayed all at once, like on a mobile interface.

## Elements

![Elements](/assets/3_components/accordion/image-20200810110853155.png)

### Title (1)

The title should be as brief as possible while being clear and descriptive.

#### Icon (2)

Indicates if the panel is open or closed.

#### Body Copy (3)

Content inside of a section may be split into paragraphs and include sub-headers and images if needed. Excessive complexity should, however, be avoided.

## Related Components

Tabs



