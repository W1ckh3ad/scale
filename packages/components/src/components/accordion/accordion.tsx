import { Element, Component, h, Prop, Host, Listen } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';
import { CssInJs } from '../../utils/css-in-js';
import { styles } from './accordion.styles';
import { Collapsible } from '../collapsible/collapsible';

@Component({
  tag: 'scale-accordion',
  shadow: false,
})
export class Accordion implements Base {
  @Element() el: HTMLElement;
  @Prop() styles?: StyleSheet;
  @CssInJs('Accordion', styles) stylesheet: StyleSheet;
  @Prop() headline?: string;

  @Prop() dependend: boolean = false;

  @Listen('toggler')
  collapsibleHandler(event: CustomEvent<Collapsible>) {
    if (this.dependend) {
      const current = event.target;
      const children = this.el.querySelectorAll('scale-collapsible');
      Array.from(children).forEach(child => {
        // tslint:disable-next-line: no-unused-expression
        child !== current && child.close();
      });
    }
  }

  @Listen('toggleHead')
  collapsibleKeyHandler(event: CustomEvent) {
    const current = event.target;
    const children = this.el.querySelectorAll('scale-collapsible');

    switch (event.detail.key) {
      case `ArrowDown`:
        Array.from(children).forEach((child, i) => {
          if (child === current) {
            i === children.length - 1
              ? children[0].setFocus()
              : children[i + 1].setFocus();
          }
        });
        break;

      case `ArrowUp`:
        event.preventDefault();
        Array.from(children).forEach((child, i) => {
          if (child === current) {
            i === 0
              ? children[children.length - 1].setFocus()
              : children[i - 1].setFocus();
          }
        });
        break;

      case `Home`:
        children[0].setFocus();
        break;

      case `End`:
        children[children.length - 1].setFocus();
        break;
    }
  }

  componentWillLoad() {}

  componentDidLoad() {}
  componentDidUnload() {}
  componentWillUpdate() {}
  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(classes['accordion']);
  }

  render() {
    const { classes } = this.stylesheet;

    return (
      <Host>
        <div class={classes.accordionContainer}>
          {/* {this.headline ? (
            <scale-text tag="h3" variant="h3">
              {this.headline}
            </scale-text>
          ) : null} */}

          <div class="accordion-container">
            <slot />
          </div>
        </div>
      </Host>
    );
  }
}
