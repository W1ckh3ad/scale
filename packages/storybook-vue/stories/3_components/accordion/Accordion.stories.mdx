import {
  Meta,
  Story,
  ArgsTable,
  Canvas,
  Description,
} from "@storybook/addon-docs/blocks";
import ScaleAccordion from "./ScaleAccordion.vue";

<Meta
  title="Components/Accordion"
  component={ScaleAccordion}
  argTypes={{
    dependent: {
      defaultValue: false,
      control: { type: "boolean" },
    },
    expanded: {
      defaultValue: false,
      control: { type: "boolean" },
    },
  }}
/>

export const Template = (args, { argTypes }) => ({
  components: { ScaleAccordion },
  props: {
    ...ScaleAccordion.props,
  },
  template: `
    <scale-accordion
      :styles="styles"
      :dependent="dependent"
      :expanded="expanded"
    >
      <scale-collapsible>
        <h3>Leo integer malesuada nunc vel risus</h3>
        <p>Freegan kinfolk <scale-link href="/">farm-to-table humblebrag cred</scale-link>, hammock bespoke small batch pabst. 90's tumblr whatever direct trade, organic master cleanse copper mug schlitz palo santo bushwick ethical pop-up chambray portland. Sartorial austin iceland street art, pug asymmetrical marfa mustache mumblecore. Shoreditch raclette knausgaard, swag enamel pin food truck everyday carry 3 wolf moon.</p>
      </scale-collapsible>
      <scale-collapsible>
        <h3>Dolor purus non enim</h3>
        <p>Bespoke austin pork belly yuccie pop-up. <scale-link href="/">Before they sold out</scale-link> YOLO kickstarter scenester meggings echo park aesthetic. Thundercats post-ironic wayfarers microdosing etsy hashtag seitan photo booth bitters.</p>
      </scale-collapsible>
      <scale-collapsible>
        <h3>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam</h3>
        <p>Biodiesel chia af hoodie tumeric bespoke letterpress man bun fashion axe helvetica brunch godard cray viral prism. Street art tattooed bitters, ugh four loko selfies you probably haven't heard of them locavore bushwick. Tattooed 90's kinfolk, banh mi umami banjo palo santo cliche. Cray wolf godard skateboard celiac taxidermy tacos offal.</p>
      </scale-collapsible>
    </scale-accordion>
  `,
});

<div style={{display: 'inline-flex', alignItems: 'center', justifyContent: 'space-between', width: '100%'}}>
    <h1>Accordion</h1>
    <img src="assets/aa.png" alt="Accessible AA" />
</div>

## Standard

<Canvas withSource="none">
  <Story name="Standard">{Template.bind({})}</Story>
</Canvas>

<ArgsTable story="Standard" />

```html
<scale-accordion>
  <scale-collapsible>
    <h3>Leo integer malesuada nunc vel risus</h3>
    <p>Freegan kinfolk farm-to-table humblebrag cred, hammock bespoke small batch pabst…</p>
  </scale-collapsible>
  <scale-collapsible>
    <h3>Dolor purus non enim</h3>
    <p>Bespoke austin pork belly yuccie pop-up…</p>
  </scale-collapsible>
  <scale-collapsible>
    <h3>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam</h3>
    <p>Biodiesel chia af hoodie tumeric bespoke letterpress…</p>
  </scale-collapsible>
</scale-accordion>
```

## Dependent

<Canvas withSource="none">
  <Story
    name="Dependent"
    args={{
      dependent: true,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-accordion dependent>
  <scale-collapsible>
    <h3>Leo integer malesuada nunc vel risus</h3>
    <p>Freegan kinfolk farm-to-table humblebrag cred, hammock bespoke small batch pabst…</p>
  </scale-collapsible>
  <scale-collapsible>
    <h3>Dolor purus non enim</h3>
    <p>Bespoke austin pork belly yuccie pop-up…</p>
  </scale-collapsible>
  <scale-collapsible>
    <h3>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam</h3>
    <p>Biodiesel chia af hoodie tumeric bespoke letterpress…</p>
  </scale-collapsible>
</scale-accordion>
```


## Expanded

<Canvas withSource="none">
  <Story
    name="Expanded"
    args={{
      expanded: true,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-accordion expanded>
  <scale-collapsible>
    <h3>Leo integer malesuada nunc vel risus</h3>
    <p>Freegan kinfolk farm-to-table humblebrag cred, hammock bespoke small batch pabst…</p>
  </scale-collapsible>
  <scale-collapsible>
    <h3>Dolor purus non enim</h3>
    <p>Bespoke austin pork belly yuccie pop-up…</p>
  </scale-collapsible>
  <scale-collapsible>
    <h3>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam</h3>
    <p>Biodiesel chia af hoodie tumeric bespoke letterpress…</p>
  </scale-collapsible>
</scale-accordion>
```
