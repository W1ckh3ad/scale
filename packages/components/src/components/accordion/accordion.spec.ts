import { newSpecPage } from '@stencil/core/testing';
import { Accordion } from './accordion';

describe('Accordion', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Accordion],
      html: `<scale-accordion></scale-accordion>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
