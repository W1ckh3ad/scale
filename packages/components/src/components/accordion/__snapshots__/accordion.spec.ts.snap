// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should match snapshot 1`] = `
<scale-accordion>
  <mock:shadow-root>
    <div class="accordion" part="base">
      <slot></slot>
    </div>
  </mock:shadow-root>
</scale-accordion>
`;
